tst/EnergyPlus/unit/IceThermalStorage.unit.cc

```diff
@@ -0,0 +1,180 @@
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// EnergyPlus::Low Temperature Radiant Unit Tests
+
+// Google Test Headers
+#include <gtest/gtest.h>
+
+// EnergyPlus Headers
+#include "Fixtures/EnergyPlusFixture.hh"
+#include <EnergyPlus/IceThermalStorage.hh>
+#include <EnergyPlus/CurveManager.hh>
+#include <EnergyPlus/UtilityRoutines.hh>
+
+#include <EnergyPlus/General.hh>
+
+using namespace EnergyPlus;
+using namespace EnergyPlus::IceThermalStorage;
+using namespace ObjexxFCL;
+using namespace DataGlobals;
+using namespace CurveManager;
+using namespace EnergyPlus::General;
+
+class IceThermalStorageTest : public EnergyPlusFixture
+{
+public:
+    int TotDetailedIce;
+
+protected:
+    virtual void SetUp()
+    {
+        EnergyPlusFixture::SetUp(); // Sets up the base fixture first.
+
+    }
+
+    virtual void TearDown()
+    {
+        EnergyPlusFixture::TearDown(); // Remember to tear down the base fixture after cleaning up derived fixture!
+    }
+};
+
+TEST_F(EnergyPlusFixture, IceThermalStorage_CalcQstarTest)
+{
+    int TotDetailedIce = 4;
+    int TotCurves = 4;
+    int IceStorageCurveType;
+    int TestNum;
+    Real64 CurveAnswer = 0.0;
+    Real64 ExpectedValue = 0.0;
+    Real64 Tolerance = 0.001;
+    
+    if (allocated(IceThermalStorage::DetIceStor)) DetIceStor.deallocate();
+    if (allocated(CurveManager::PerfCurve)) PerfCurve.deallocate();
+    
+    IceThermalStorage::DetIceStor.allocate(TotDetailedIce);
+    CurveManager::PerfCurve.allocate(TotCurves);
+    CurveManager::NumCurves = TotCurves;
+    BeginEnvrnFlag = false;
+    
+    //Test 1: CurveVarsFracChargedLMTD Curve is QuadraticLinear
+    TestNum = 1;
+    IceStorageCurveType = 1; // FracChargedLMTD
```

Comment:
Nice unit test. It would be more robust and readable to use the actual parameter variables to assign `IceStorageCurveType` and `PerfCurve.CurveType`.

e.g. `IceStorageCurveType = IceThermalStorage::CurveVarsFracChargedLMTD`.

Reply:
@mjwitte Took care of this as well. Will make the commits and if everything comes back as before (hopefully the case), this will be ready for re-review.